wayland-protocols/ext-lockscreen_v1.xml

```diff
@@ -0,0 +1,104 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<protocol name="ext-lockscreen_v1">
+  <copyright>    
+    Copyright © Canonical Ltd.
+
+    This program is free software: you can redistribute it and/or modify it
+    under the terms of the GNU General Public License version 2 or 3,
+    as published by the Free Software Foundation.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+    
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+  </copyright>
+  <description summary="compositor authoritative screen-locking mechanism">
+    This protocol allows for a priveleged Wayland client to display arbitrary graphics
+    on the screen when the compositor informs the client that the session has been
+    locked.
+
+    The compositor may choose to restrict this protocol to a special client
+    launched by the compositor itself or expose it to all privileged clients.
+    This is up to the compositor policy.
+
+    The compositor is responsible for deciding when the session should be locked and unlocked.
+    If the client dies while the session is locked, the client may simply resubmit
+    a buffer for display at a later time. During this time, the compositor itself
+    may have been unlocked via some other mechanism (e.g. loginctl). Exactly how and
+    when the compositor locks and unlocks is the policy of the compositor.
+
+    A typical use case is for the compositor to lock or unlock when it receives a
+    signal from Logind to do so.
+
+    The key words "must", "must not", "required", "shall", "shall not",
+    "should", "should not", "recommended",  "may", and "optional" in this
+    document are to be interpreted as described in IETF RFC 2119.
+  </description>
+
+  <interface name="ext_lockscreen_manager_v1" version="1">
+    <description summary="used to create lockscreen surfaces">
+      
+    </description>
+    <request name="associate">
+      This request creates a lockscreen surface and asks the compositor to show this
+      surface on the provided output when the session is next locked. If the session is
+      already locked, the compositor will display this surface right away.
```

Comment:
Just checking I've understood correctly: the client needs to track outputs being added, removed and reconfigured in order to create and associate the corresponding surfaces and buffers _regardless of whether the session is locked_.

Is there a reason not to create lockscreen content "on demand"?

Reply:
> Is there a reason not to create lockscreen content "on demand"?

The reason we're thinking of is visual feedback - if the buffers are "preloaded", the compositor can slot them in immediately, rather than waiting for the lockscreen to "warm up".

Yes, the trade off is higher memory use, so we should support both modes.

Reply:
Where do you (not) see it? Or you mean you don't expect it to be noticable?

I mean, I suppose the app can have the surface pre-cooked without actually pushing a buffer through - in which case, yeah I agree that shouldn't be noticable.